geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewaySenderCommand.java

```diff
@@ -75,6 +75,11 @@ public ResultModel createGatewaySender(
           mandatory = true,
           help = CliStrings.CREATE_GATEWAYSENDER__REMOTEDISTRIBUTEDSYSTEMID__HELP) Integer remoteDistributedSystemId,
 
+      @CliOption(key = CliStrings.CREATE_GATEWAYSENDER__GROUPTRANSACTIONEVENTS,
+          specifiedDefaultValue = "true",
+          unspecifiedDefaultValue = "false",
+          help = CliStrings.CREATE_GATEWAYSENDER__GROUPTRANSACTIONEVENTS__HELP) boolean groupTransactionEvents,
+
```

Comment:
This new option should be added to the docs for the create gateway-sender gfsh command in `create.html.md.erb`.

Reply:
Yep

Reply:
Note that the GatewaySender can be configured via the Java API ... not just through gfsh.  This is where I was hoping to get that check to ensure the user is configuring a valid GatewaySender (Serial GatewaySender).

Also, during internal testing with HA (killing servers) we've noticed data inconsistency between WAN Sites with groupTransactionEvents enabled.  In addition, I've seen the primary Serial GatewaySender AEQ not drain (again, with this attribute enabled).  

Do you have any DUnit Tests with HA (to verify that we still have consistency between wan sites and queues being drained when groupTransactionEvents is enabled)?

We are still investigating this and will update this PR as we narrow in on the root cause.

Reply:
I forgot to add the check for the creation of the GatewaySender via the Java API.
I have just pushed a new commit with this check in `GatewaySenderFactoryImpl.`

Regarding your question about DUnit Tests with HA, I have not written any. I will try to investigate from my side the problems you have encountered.

Reply:
I ran both `ParallelWANStatsDUnitTest::testParallelPropagationHA()` and `SerialWANStatsDUnitTest::testReplicatedSerialPropagationHA()` but setting group-transaction-events to true in the gateway-senders used and both test cases passed too.

Reply:
> Note that the GatewaySender can be configured via the Java API ... not just through gfsh. This is where I was hoping to get that check to ensure the user is configuring a valid GatewaySender (Serial GatewaySender).
> 
> Also, during internal testing with HA (killing servers) we've noticed data inconsistency between WAN Sites with groupTransactionEvents enabled. In addition, I've seen the primary Serial GatewaySender AEQ not drain (again, with this attribute enabled).
> 
> Do you have any DUnit Tests with HA (to verify that we still have consistency between wan sites and queues being drained when groupTransactionEvents is enabled)?
> 
> We are still investigating this and will update this PR as we narrow in on the root cause.

I think I found the problem. There was a bug in the handling of events that had to be added to the batch in order to complete a transaction that under high load and concurrency situations caused that some events were not propagated.
I have added DUnit tests with HA for the serial and gateway senders. The one for the SerialGatewaySender was failing before my fix.

Please, let me know if the new commit fixes the problem you encountered.

Reply:
@ladyVader Any news after the data inconsistency you found after having pushed my fix?

Reply:
Thanks for both the new Exception on misconfiguring the SerialGatewaySender with groupTransactionEvents enabled + dispatcherThreads > 1.

In addition, I'm no longer seeing the data inconsistencies or hangs waiting for the primary gateway sender queue to drain.